test/upgrade/probe.go

```diff
@@ -51,7 +51,7 @@ func ProbeTest() pkgupgrade.BackgroundOperation {
 			// This polls until we get a 200 with the right body.
 			assertServiceResourcesUpdated(c.T, clients, *names, url, test.PizzaPlanetText1)
 
-			prober = test.RunRouteProber(c.T.Logf, clients, url, test.AddRootCAtoTransport(context.Background(), c.T.Logf, clients, test.ServingFlags.HTTPS))
+			prober = test.RunRouteProber(c.Log.Infof, clients, url, test.AddRootCAtoTransport(context.Background(), c.T.Logf, clients, test.ServingFlags.HTTPS))
```

Comment:
```suggestion
			prober = test.RunRouteProber(c.Log.Infof, clients, url, test.AddRootCAtoTransport(context.Background(), c.Log.Infof, clients, test.ServingFlags.HTTPS))
```

Reply:
There is a second one there.

Reply:
This change is not necessary because the call happens in the current test context. It's only used to setup the test, not later.

Reply:
Right, but why use 2 different loggers? Logger can be configured centrally, and this feeds some messages to one logger and some to the other.

For example, someone creates the logger to write messages to a file. Then he will be surprised that not all messages are collected. Or worse, he can miss some important information.

Reply:
IMO, what happens in the current test context should be logged by the test logger. If there's a long running process it should use a different logger.
If I take the same logic and would like to use just one logger throughout the test, I'd have to change the `assertServiceResourcesUpdated` a few lines above (which would propagate to the E2E test suite and result in major refactoring), re-implement c.T.Fatal because it's logging stuff as well. It's not the goal of this PR.

Reply:
I think any part of the upgrade test should use Fatal. It should use Error at most.

Reply:
Why is that? We're re-using functions from the E2E test suites where they mostly use `t.Fatal` (it's probably for a different discussion though)